sdk/keyvault/Azure.Security.KeyVault.Secrets/tests/SecretsTests.cs

```diff
@@ -0,0 +1,259 @@
+        }
+
+        [Test]
+        public async Task CrudWithExtendedProps()
+        {
+            var client = new SecretClient(VaultUri, AzureCredential.Default);
+
+            var secret = new Secret("CrudWithExtendedProps", "CrudWithExtendedPropsValue1")
+            {
+                ContentType = "password",
+                NotBefore = UtcNowMs() + TimeSpan.FromDays(1),
+                Expires = UtcNowMs() + TimeSpan.FromDays(90)
+            };
+
+            Secret setResult = await client.SetAsync(secret);
+
+            Assert.AreEqual("password", setResult.ContentType);
+
+            Secret getResult = await client.GetAsync("CrudWithExtendedProps");
+
+            AssertSecretsEqual(setResult, getResult);
+
+            DeletedSecret deleteResult = await client.DeleteAsync("CrudWithExtendedProps");
+
+            // remove the value which is not set on the deleted response
+            typeof(Secret).GetProperty(nameof(setResult.Value)).SetValue(setResult, null);
+
+            AssertSecretsEqual(setResult, deleteResult);
+        }
+
+        [Test]
+        public async Task BackupRestore()
+        {
+            var backupPath = Path.GetTempFileName();
+
+            try
+            {
+                var client = new SecretClient(VaultUri, AzureCredential.Default);
+
+                Secret setResult = await client.SetAsync("BackupRestore", "BackupRestore");
+
+                byte[] backup = await client.BackupAsync("BackupRestore");
+
+                await client.DeleteAsync("BackupRestore");
+
+                Secret restoreResult = await client.RestoreAsync(backup);
+
+                // remove the vaule which is not set in the restore response
+                typeof(Secret).GetProperty(nameof(setResult.Value)).SetValue(setResult, null);
+
+                AssertSecretsEqual(setResult, restoreResult);
+            }
+            finally
+            {
+                File.Delete(backupPath);
+            }
+        }
+
+        private DateTime UtcNowMs()
+        {
+            return DateTime.MinValue.ToUniversalTime() + TimeSpan.FromMilliseconds(new TimeSpan(DateTime.UtcNow.Ticks).TotalMilliseconds);
+        }
+
+    }
+
+    public class SecretListTests : KeyVaultTestBase, IDisposable
+    {
+        private const int VersionCount = 50;
+        private readonly string SecretName = Guid.NewGuid().ToString("N");
+
+        private readonly Dictionary<string, Secret> _versions = new Dictionary<string, Secret>(VersionCount);
+        private readonly SecretClient _client;
+
+        public SecretListTests()
+        {
+            _client = new SecretClient(VaultUri, AzureCredential.Default);
+
+            for (int i = 0; i < VersionCount; i++)
+            {
+                Secret secret = _client.SetAsync(SecretName, Guid.NewGuid().ToString("N")).GetAwaiter().GetResult();
+
+                typeof(Secret).GetProperty(nameof(secret.Value)).SetValue(secret, null);
```

Comment:
why is it necessary to have the value set to `null`?

Reply:
Hmmm I see. Because we are creating every secret with a different value here `Guid.NewGuid().ToString("N")`?